Library/Homebrew/PATH.rb

```diff
@@ -0,0 +1,61 @@
+class PATH
+  def initialize(*paths)
+    @paths = parse(*paths)
+  end
+
+  def prepend(*paths)
+    @paths.unshift(*parse(*paths))
+    self
+  end
+
+  def append(*paths)
+    @paths.concat(parse(*paths))
+    self
+  end
+
+  def to_ary
+    @paths
+  end
+  alias to_a to_ary
+
+  def to_str
+    @paths.join(File::PATH_SEPARATOR)
+  end
+  alias to_s to_str
+
+  def eql?(other)
```

Comment:
If you implement `#eql?`, you need to implement `#hash` as well. Something like this would do:

```ruby
def hash
  [self.class, @paths].hash
end
```